[GLOBAL]
pants_version = "2.11.0rc0"
backend_packages = [
  # TODO: There is a conflict between the runtime dependencies for Python and Scala:
  #   see https://github.com/pantsbuild/pants/issues/14694#issuecomment-1083565675
  #"pants.backend.codegen.protobuf.python",
  "pants.backend.codegen.thrift.apache.python",
  "pants.backend.experimental.codegen.protobuf.scala",
  "pants.backend.experimental.java",
  "pants.backend.experimental.scala",
  "pants.backend.python",
]

[anonymous-telemetry]
enabled = true
repo_id = "205E301D-44A4-42FF-8495-C9D41871A431"

[python]
interpreter_constraints = ["CPython>=3.8,<3.11"]

# Enable the "resolves" mechanism, which turns on lockfiles for user code. See
 # https://www.pantsbuild.org/docs/python-third-party-dependencies. This also adds the
 # `generate-lockfiles` goal for Pants to generate the lockfile for you, although in some
 # situations you may need to manually generate it, as explained in the docs.
enable_resolves = true

[scala.version_for_resolve]
"jvm-default" = "2.13.8"

[scalapb]
# In Pants `2.11.x`, the runtime dependency of `scalapb` targets must be explicitly specified. But
# in `2.12.x`, it is inferred via #14898 based on its presence in the relevant thirdparty resolve.
runtime_dependencies = [
  '3rdparty/jvm/com/thesamet/scalapb:scalapb-runtime_2.13'
]
